cache/authors_stats.go

```diff
@@ -0,0 +1,76 @@
+package cache
+
+import (
+	"database/sql"
+
+	"github.com/Bnei-Baruch/sqlboiler/queries"
+	"github.com/lib/pq"
+	"github.com/pkg/errors"
+
+	"github.com/Bnei-Baruch/archive-backend/mdb"
+)
+
+type AuthorsStatsCache interface {
+	Provider
+	GetSources(codes []string) []string
+}
+
+type AuthorsStatsCacheImpl struct {
+	mdb    *sql.DB
+	byCode map[string][]string
+}
+
+func NewAuthorsStatsCacheImpl(mdbDB *sql.DB) AuthorsStatsCache {
+	stats := new(AuthorsStatsCacheImpl)
+	stats.mdb = mdbDB
+	stats.byCode = make(map[string][]string)
+	for c, _ := range mdb.AUTHOR_REGISTRY.ByCode {
+		stats.byCode[c] = make([]string, 0)
+	}
+	return stats
+}
+
+func (s *AuthorsStatsCacheImpl) String() string {
+	return "AuthorsStatsCacheImpl"
+}
+
+func (s *AuthorsStatsCacheImpl) Refresh() error {
+	err := s.load()
+	return errors.Wrap(err, "Load tags and sources stats.")
+}
+
+func (s *AuthorsStatsCacheImpl) load() error {
+	rows, err := queries.Raw(s.mdb, `
+		SELECT a.code, array_agg(DISTINCT s.uid) FROM authors_sources "as"
```

Comment:
SQL ERROR: column "a.code" must appear in the GROUP BY clause or be used in an aggregate function

Reply:
fixed